build/build-clspv.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+## $1 : version, clspv does not have any and only uses master branch.
+## $2 : destination: a directory or S3 path (eg. s3://...)
+## $3 : last revision successfully build
+
+set -ex
+
+ROOT=$PWD
+VERSION="${1}"
+LAST_REVISION="${3}"
+
+if [[ "${VERSION}" != "master" ]]; then
+    echo "Only support building master"
+    exit 1
+fi
+
+URL="https://github.com/google/clspv.git"
+BRANCH="master"
+
+REVISION=$(git ls-remote --heads "${URL}" "refs/heads/${BRANCH}" | cut -f 1)
+echo "ce-build-revision:${REVISION}"
+
+if [[ "${REVISION}" == "${LAST_REVISION}" ]]; then
+    echo "ce-build-status:SKIPPED"
+    exit
+fi
+
+FULLNAME=clspv-${VERSION}-$(date +%Y%m%d)
+
+OUTPUT=${ROOT}/${FULLNAME}.tar.xz
+S3OUTPUT=
+if [[ $2 =~ ^s3:// ]]; then
+    S3OUTPUT=$2
+else
+    if [[ -d "${2}" ]]; then
+        OUTPUT=$2/${FULLNAME}.tar.xz
+    else
+        OUTPUT=${2-$OUTPUT}
+    fi
+fi
+
+## From now, no unset variable
+set -u
+
+OUTPUT=$(realpath "${OUTPUT}")
+
+git clone --depth 1 "${URL}" --branch "${BRANCH}"
+pushd clspv
+
+python3 utils/fetch_sources.py
+
+mkdir build
+cd build
+
+cmake -G Ninja ..
+ninja
```

Comment:
It would be slightly cleaner to use `cmake --build` instead of running `ninja` directly.

The code could be reduced to three lines.  Something like:
```
mkdir build
cmake -S . -B build -G "Ninja"
cmake --build build
```

Using `-G "Unix Makefiles"` instead would allow the dependency on Ninja to be removed.

(Note that even when running `ninja` directly, it accepts a `-C` option that specifies the directory to run in, so we can still avoid changing directory with `cd` in the script, if we prefer not to.)

Reply:
Thanks. I used Ninja since that's what is recommended in the clspv repo, but this makes sense too.